src/App.svelte

```diff
@@ -40,6 +40,15 @@
   mapsStore.set(settings.maps.map((map, index) => ({ ...map, index })));
   stylePresetsStore.set(settings.stylePresets);
 
+  // Since maps comes from settings, only reset as needed
+  let maps = [];
+  $: if (JSON.stringify(maps) !== JSON.stringify(settings.maps)) {
+    maps = settings.maps;
+  }
```

Comment:
We have two instances where we do this, and I wonder if we should have a little utility function we call to handle objects like this when necessary rather than write this out twice?

Reply:
If you mean specifically the deep-equal comparison that sounds good to me, or use a library with such a utility function.